cid/helpers/quicksight/__init__.py

```diff
@@ -256,9 +261,23 @@ def _safe_int(value, default=None):
                     logger.info(f'Unable to describe template for {dashboardId}, {exc}')
             else:
                 logger.info("Minimum template version could not be found for Dashboard {dashboardId}: {_template_arn}, deployed template could not be described")
+        
+        if _template_arn is None:
+            # Resolve source definition (the latest definition publicly available)
+            data_stream = io.StringIO(_definition["data"])
+            definition_data = yaml.safe_load(data_stream)
+            dashboard.sourceDefinition = CidQsDefinition(definition_data)
```

Comment:
1/ _definition["data"] can be a dict or string
2/ both theoretically can be templates as per this code:

https://github.com/aws-samples/aws-cudos-framework-deployment/blob/d2865cb42e50ff836bd44d9ce635a789d3605536/cid/common.py#L482C1-L489C62

so ideally would be great to have it as one piece of code. I do not see how it can be done simply on the level of helper quicksight as parameters on the common.py.